types/aws-lambda/trigger/api-gateway-proxy.d.ts

```diff
@@ -133,10 +133,23 @@ export interface APIGatewayProxyEventV2 {
         accountId: string;
         apiId: string;
         authorizer?: {
-            jwt: {
+            jwt?: {
```

Comment:
Oof. Breaking change, but it's hard to fix. Let me think about this.

Reply:
I think if we're breaking people, it would be nice if the fix actually made things nicer for the broken person, by not making them need to assert the authorizer context any more. The v1 version of this event has what seems to be the solution to the exact same situation, so something like that would be nice.

As a more minor note: please add some tests, and use the formatter (check the docs prettier)

Reply:
Hi, 
Just to be sure I understood you, you mean changing authorizer definition to be that object or ```null```?
Regarding tests I'm really sorry but I read tests from api-gateway.tests.ts and didn't really understood them. Could you indicate any of them that I can use as template to develop a new one?
Thanks,

Reply:
> changing authorizer definition to be that object or null?

This doesn't really make sense? Take a look at how the V1 stuff actually works with authorizer, essentially the caller passes the authorizer content shape as a generic parameter to the various types, because they are in control of that, and we give some common types.

Regarding tests, you want them to be in `types/aws-lambda/test/api-gateway-tests.ts`, which is very messy at the moment as it's testing all of the gory types I mention above. Normally it would just be adding a few lines in `proxyHandlerV2` to check it's accessible, as you are adding to the V2 event, but it seems there are no tests for the current authorizer context tests. Ideally, you would be adding lines that would be roughly what you expect the user to be doing to use the things you're adding. At the moment, that might be:

```ts
    str = event.requestContext.authorizer!.iam!.userArn;
```

But the `!` is a signal this isn't really what you want to be doing as the design: the user should *know* they are going to have this. This is why the current v1 types are tests are so complicated.

Reply:
I changed the definition following the commit by @sandersn but after carefully reviewing  ```types/aws-lambda/test/api-gateway-tests.ts``` I declare myself incapable of creating new tests

Reply:
@simonbuchan unless you have time to suggest some test cases, I think this PR is blocked.